narwhals/_arrow/series.py

```diff
@@ -178,6 +178,14 @@ def __rtruediv__(self, other: Any) -> Self:
             other = pa.scalar(other)
         return self._from_native_series(pc.divide(*cast_for_truediv(other, ser)))
 
+    def __mod__(self, other: Any) -> Self:
+        pc = get_pyarrow_compute()
+        ser = self._native_series
+        other = validate_column_comparand(other)
+        floor_div = (self // other)._native_series
+        res = pc.subtract(ser, pc.multiply(floor_div, other))
+        return self._from_native_series(res)
```

Comment:
Actually, this could be written very beautify as:

```suggestion
    def __mod__(self, other: Any) -> Self:
        other = validate_column_comparand(other)
        return self - self // other * other
```